src/core/server/i18n/get_kibana_translation_files.ts

```diff
@@ -17,26 +17,27 @@
  * under the License.
  */
 
-import { KibanaConfig } from '../kbn_server';
-import { fromRoot } from '../../../core/server/utils';
-import { I18N_RC } from './constants';
+import { basename } from 'path';
+import { fromRoot } from '../utils';
 import { getTranslationPaths } from './get_translation_paths';
 
-export async function getKibanaTranslationPaths(config: Pick<KibanaConfig, 'get'>) {
-  return await Promise.all([
+export const getKibanaTranslationFiles = async (
+  locale: string,
+  pluginPaths: string[]
+): Promise<string[]> => {
+  const translationPaths = await Promise.all([
     getTranslationPaths({
       cwd: fromRoot('.'),
-      glob: `*/${I18N_RC}`,
+      nested: true,
     }),
```

Comment:
AFAIK, this is only used to get the `x-pack/.i18nrc.json` file. However to avoid any risk / breaking changes (User could have custom i18nrc files in another non-plugin folder eventually), I kept it as it was done in legacy.

Reply:
@pgayvallet only in `x-pack`. I believe it is safe to change it as long as `x-pack` and kibana's root dir are discoverable.

Reply:
does it mean that it traverses all the `node_modules` and other unnecessary folders?

Reply:
No. `*/something`  globbing is only scanning a single level (as opposed to `**/something`)

Reply:
ok, I didn't check implementation. but `nested: true` traversing on a single level in-depth is confusing 😅

Reply:
Yea, you're right. I kinda wanted to rename it to `includeSubFolders`, but is was way longer, and not really more explicit that we are only scanning one level down. No issue renaming if you have a better idea.

Reply:
We will need something similar to `maxScanDepth` in discovery to support migration to Domain-based folder structure. But I'm okay to keep it as is for now.